scripts/modules_auto_pr/modules_auto_pr.sh

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+
+set -e  # Exit immediately if a command exits with a non-zero status
+set -x  # Print commands and their arguments as they are executed
+
+####################################################################################################
+#                                            VARIABLES                                             #
+####################################################################################################
+
+DATE=$(date +"%Y-%m-%d_%Hh%M")
+REPO_URL="git@github.com:hpcugent/vsc_user_docs.git"
+BASE_BRANCH="main"
+BRANCH_NAME="auto_update_modules_$DATE"
+REPO_NAME="vsc_user_docs" # script available_modules.py requires this to be the name. Do not change.
+REPO_PATH="$VSC_SCRATCH/modules_auto_pr_script_$DATE/$REPO_NAME"
+COMMIT_MESSAGE="Update Modules"
+PR_TITLE="Auto Update Modules"
+
+make_pr_body() {
+  local n_added_modules="$1"
+  local n_removed_modules="$2"
+
+  echo "This is an automated pull request updating the markdown files of all available modules."
+  echo ""
+  echo "Changes:"
+  echo "- Updated the markdown files of all previously available modules"
+  echo "- Added $n_added_modules new modules"
+  echo "- Removed $n_removed_modules modules"
+}
+
+####################################################################################################
+#                                             SCRIPT                                               #
+####################################################################################################
+
+echo_info()  { echo -e "\e[32m$0: [INFO] $1\e[0m"; }
+echo_error() { echo -e "\e[31m$0: [ERROR] $1\e[0m"; }
+
+main() {
+  local github_pat_file="$1"
+
+  # Check if the GitHub Personal Access Token file exists
+  if [ ! -f "$github_pat_file" ]; then
+      echo_error "GitHub Personal Access Token file not found."
+      echo_error "Please provide the path to the file containing the GitHub Personal Access Token."
+      echo_error "Usage: $0 <github_pat_file>"
+      exit 1
+  fi
+
+  gh auth login --with-token < "$github_pat_file"
+
+  # Clone the repository and create a new branch
+  git clone $REPO_URL "$REPO_PATH"
```

Comment:
Make sure this is done in a unique subdirectory of `/tmp/$USER`, `mktemp -d` is your friend here

And clean up the tmpdir at the end